Use TaskType keys for counting tasks instead of hardcoding the key values

costinsin opened the pull request:
Change the type of `_taskCounts` to an IndexSignature that can have keys named as the values of the `TaskType` type.

The Closure Compiler used at Google has a property renaming optimization that can change the property names when minifying code. Having the correct type helps the TSJS team that develops a tool to identfy property renaming issues directly in TypeScript.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [x] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the new behavior?
The program behavior did not change.


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->

JiaLiPassion reviewed (approved):
LGTM! Thanks for the PR.

JeanMeche commented:
Closing as this PR is stale and we're looking to reduce the number of open PR. Thank you for your understanding.

costinsin commented:
Maybe it would have been easier to close it 1 year ago by merging it. It was already approved by then. Have a great day!

JeanMeche commented:
If you'd like the rebase it, we can re-open the PR. It was probably forgotten to add it to the merge queue. Because of the synchronization between github and Google's monorepo the merge is less trivial than usual.

costinsin committed:
refactor(zone.js): Change the type of `_taskCounts` to an IndexSignature that can have keys
named as the values of the `TaskType` type.

The Closure Compiler used at Google has a property renaming optimization
that can change the property names when minifying code. Having the
correct type helps the TSJS team that develops a tool to identfy
property renaming issues directly in TypeScript.

Signed-off-by: Costin Sin <sin.costinrobert@gmail.com>

costinsin commented:
Hello! Sorry for the delay. I have rebased the change now as you asked me.

costinsin commented:
Hey @JeanMeche can you take a look over this PR, please?

costinsin commented:
I have signed the CLA.

thePunderWoman commented:
This PR was merged into the repository by commit 2e54d6dea843203a7361d9a983d65b682d9b2797.

The changes were merged into the following branches: main, 18.2.x, 19.0.x
